docs/source/development/areas/annotate_package.rst

```diff
@@ -0,0 +1,205 @@
+                  - to_json
+              side_effects:
+                - mutated_value:
+                    external_state: file_system
+
+    In fact, since the two items share much in common (the only difference is in ``class_method_names``),
+    we can condense the above annotation to the following:
+
+    .. code-block:: yaml
+        :emphasize-lines: 9
+
+        # pandas.annotations.yaml
+
+        - module: pandas.core.generic
+          annotations:
+            - criteria:
+                class_instance: NDFrame
+                class_method_names:
+                  - to_csv
+                  - to_json
+              side_effects:
+                - mutated_value:
+                    external_state: file_system
+
+    where we simply extend ``class_method_names`` list instead.
+
+Annotating APIs that modify the caller itself
+---------------------------------------------
+
+Another group of APIs that need annotation are those that modify their caller.
+
+For instance, without relevant package annotation, running
+
+.. code-block:: python
+
+    from sklearn.linear_model import LinearRegression
+    import pandas as pd
+    import lineapy
+
+    df = pd.read_csv("https://raw.githubusercontent.com/LineaLabs/lineapy/main/examples/tutorials/data/iris.csv")
+    mod = LinearRegression()
+    mod.fit(X=df[["petal.width"]], y=df["sepal.width"])
+    artifact = lineapy.save(mod, "fitted_model")
+
+    print(artifact.get_code())
+
+would display
+
+.. code-block:: none
+
+    from sklearn.linear_model import LinearRegression
+
+    mod = LinearRegression()
+
+where the critical step of model fitting is missing. This is because ``lineapy`` does not recognize
+`fit <https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html#sklearn.linear_model.LinearRegression.fit>`_
+as an API that modifies the caller (``mod`` of ``LinearRegression`` type, in this case) and hence skips it
+during the relevant code cleanup.
+
+In contrast, we would get the correct code cleanup with the following package annotation (currently in the source code):
+
+.. code-block:: yaml
+
+    # sklearn.annotations.yaml
+
+    - module: sklearn.base
+      annotations:
+        - criteria:
+            class_instance: BaseEstimator
+            class_method_name: fit
+          side_effects:
+            - mutated_value:
+                self_ref: SELF_REF
+            - views:
+                - self_ref: SELF_REF
+                - result: RESULT
+
+where ``views`` specifies entities whose modifications should be linked to one another. In this example,
+``result`` and ``self_ref`` are set to be views of each other, which means that modification of the ``fit``
+call's output (e.g., ``fitted_mod`` in ``fitted_mod = mod.fit(...)``) will modify the ``fit`` caller itself
+(e.g., ``mod`` in ``fitted_mod = mod.fit(...)``), and vice versa. Differently put, ``views`` establishes
+links between objects that need to be tracked together.
```

Comment:
Add "because a change to one will change to other." to explain why we want views.